task_DEV_4/task_DEV_4/Parallelogram.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+
+namespace task_DEV_4
+{
+    class Parallelogram : Square
+    {
+        public double Height { get; set; }
+        public double Angle { get; set; }
+        public double SecondSide { get; set; }
+
+        public Parallelogram(double firstSide, double height) : base(firstSide)
+        {
+            Height = height;
+        }
+
+        public Parallelogram(double firstSide, double secondSide, double angle) : base(firstSide)
+        {
+            SecondSide = secondSide;
+            Angle = angle;
+        }
+
+        public override double CalculateArea()
+        {
+            if (Height > 0 && FirstSide > 0)
+            {
+                return FirstSide * Height;
+            }
+            else if (FirstSide > 0 && SecondSide > 0 && Angle > 0)
+            {
+                return FirstSide * SecondSide * Math.Sin(Angle);
+            }
+            else
+            {
+                throw new ArgumentOutOfRangeException();
```

Comment:
It would be more prefferable to provide your message to the constructor of exception smth like:
`throw new ArgumentOutOfRangeException("Your message");`
Then You can do in catch block smth like this
```
catch (ArgumentOutOfRangeException e)
{
    Console.WriteLine(e.Message);
}
```

Reply:
Done